test/fluree/db/transact/update_test.clj

```diff
@@ -120,3 +120,202 @@
                               '{:select {?s [:*]}
                                 :where  [[?s :id :ex/jane]]}))
             "Jane's age should now be updated to 31 (from 30).")))))
+
+(deftest transaction-functions
+  (let [conn @(fluree/connect {:method :memory})
+        ledger @(fluree/create conn "functions" {:defaultContext [test-utils/default-str-context
+                                                                  {"ex" "http://example.com/"}]})
+        db0 (fluree/db ledger)
+        db1 @(fluree/stage db0 [{"id" "ex:create-predicates"
```

Comment:
What does this `db1` step do? It looks like you're staging all the data you need for each test below, and we should be creating predicates on the fly as data is staged.

Reply:
That's an artifact of how I originally wrote the test harnesses, I can move the predicate creation into the relevant test cases.